db-tests/src/test/java/gov/nasa/jpl/aerie/database/ExternalEventTests.java

```diff
@@ -0,0 +1,2210 @@
+
+        // create a Derivation Group
+        insertDerivationGroup(DERIVATION_GROUP, SOURCE_TYPE);
+
+        // create a source
+        insertExternalSource(src);
+
+        // delete the source type (expect error)
+        final SQLException ex = assertThrows(PSQLException.class, () -> statement.executeUpdate(
+            // language=sql
+            """
+            DELETE FROM merlin.external_source_type WHERE name='%s';
+            """.formatted(SOURCE_TYPE)
+          )
+        );
+        if (!ex.getSQLState().equals("23503")
+            && !ex.getMessage().contains(
+                "update or delete on table \"external_source_type\" violates foreign key constraint "
+                + "\"derivation_group_references_external_source_type\" on table \"derivation_group\"")
+        ) {
+          throw ex;
+        }
+      }
+    }
+
+    /**
+     * It should not be possible to delete an event type with an event still associated.
+     */
+    @Test
+    void deleteEventTypeWithRemainingEvent() throws SQLException {
+      try (final var statement = connection.createStatement()) {
+
+        // create source and event
+        ExternalSource src = new ExternalSource(
+            "A",
+            SOURCE_TYPE,
+            DERIVATION_GROUP,
+            "2024-01-01T00:00:00Z",
+            "2024-01-01T00:00:00Z",
+            "2024-01-01T00:30:00Z",
+            CREATED_AT
+        );
+        ExternalEvent evt = new ExternalEvent("A_1", "2024-01-01T00:00:00Z", "00:05:0", src);
+
+        // insert the event and her types
+        // insert generic external event type, source type, and derivation group
+        insertStandardTypes();
+
+        // insert sources
+        insertExternalSource(src);
+
+        // insert events
+        insertExternalEvent(evt);
+
+        // delete the event type (expect error)
+        final SQLException ex = assertThrows(PSQLException.class,
+          () -> statement.executeUpdate(
+              // language=sql
+              """
+              DELETE FROM merlin.external_event_type WHERE name='%s';
+              """.formatted(EVENT_TYPE)
+          )
+        );
+        if (!ex.getSQLState().equals("23503")
+            && !ex.getMessage().contains(
+                "update or delete on table \"external_event_type\" violates foreign key constraint "
+                + "\"external_event_references_event_type_name\" on table \"external_event\"")
+        ) {
+          throw ex;
+        }
+      }
+    }
+  }
+  //endregion
+
+
+  //region Miscellaneous
+  /**
+   * Finally, we include an extra test. This is a comprehensive test for derived events that manages several derivation
+   *    groups (which entails repeating basicDerivedEvents twice but the second round bears a new DG name. Then, we
+   *    verify that there is no overlap! Note that this test is effectively vacuous but is a good sanity check.).
+   */
+  @Test
+  void superDerivedEvents() throws SQLException {
```

Comment:
This test should be in the `DerivationTests` section of the tests, as it testing derivation behavior.